rest/auth.py

```diff
@@ -311,6 +305,37 @@ def inner(*args, **kwargs):
 
     return decorator
 
+
+# Retrive current user role and login
+def get_current_user():
+    if not auth_enabled:
+        return abort(403, "Authentication has \
+            to be enabled to retrieve current user")
+
+    token = retrieve_token()
+    # No exception is thrown from this method but return None when failed
+    try:
+        user = User.verify_auth_token(token)
+    except AuthenticationError:
+        return abort(403, "authentication failed")
+    except AllUnauthorizedError:
+        return abort(403, "access forbidden to role all")
+
+    if user is None:
+        return abort(403, "authentication failed")
+    return user
+
+
+def retrieve_token():
+    if 'Authorization' not in request.headers:
```

Comment:
Please add docstring to explain what this function does, notably in which case it aborts.